app/assets/javascripts/authentication-complete.js

```diff
@@ -0,0 +1,17 @@
+(function() {
+  const authenticationData = JSON.parse(
+    document.getElementById("data-authentication").dataset.authenticationData
+  );
+
+  Discourse.showingSignup = true;
+  require("discourse/routes/application").default.reopen({
+    actions: {
+      didTransition: function() {
+        Em.run.next(function() {
+          Discourse.authenticationComplete(authenticationData);
```

Comment:
Tested this locally with the Linkedin OAuth plugin with [`full_screen_login: true`](https://github.com/discourse/discourse/blob/a515ba86120d605cdd5b77f36ecff94169dd229a/app/controllers/users/omniauth_callbacks_controller.rb#L65).